### PR TITLE
Swarm 324: Bootstrap performance issues

### PR DESCRIPTION
**Pull request by heiko-braun**
This is for review purposes only.

Moving to persistent jandex indexes

### CODE REVIEW

**Comment by heiko-braun** [comment]
@kenfinnigan Here's a working example for wfs-core. Maybe we can use it as a baseline for discussing the changes needed to enable persistent indexes?

**Comment by heiko-braun** [comment]
@kenfinnigan Please not that it requires these changes to be merged into the jandex plugin. Hence the 1.0.4-SNAPSHOT dependency which causes the CI build to fail.

https://github.com/wildfly/jandex-maven-plugin/pull/9

**Comment by kenfinnigan** [comment]
FYI, I had added the jandex plugin to parent pom so it can be shared across more than core.

Is there any benefit to customizing the index name? We're not doing anything WildFly Swarm specific in building the jandex right?

**Comment by heiko-braun** [comment]
The custom index name is necessary, because otherwise it conflicts with indexes that WF expects at that location.

**Comment by kenfinnigan** [comment]
Is it building differently than what WF expects? Otherwise I would not expect it to be an issue

**Comment by heiko-braun** [comment]
yes, the indexes seem to be different. i didn't have the time to fully track down all the inner workings of deployer chains in WF and how they leverage jandex, but I can tell that our integration tests fail with indexes provided by us that carry the default name

**Comment by heiko-braun** [comment]
The latest update pushes the jandex plugin into the fractions. If this passes the tests, then we would be ready to apply the changes to the other fraction outside of core

**Comment by heiko-braun** [comment]
The latest commit removes the jandex file from main/module.xml

**Comment by heiko-braun** [comment]
Cleaned up and squashed the commits to make it more comprehensible

**Comment by heiko-braun** [comment]
With this build setup (jandex in the core pom) all our own fractions that don't reside within core inherit everything they need to make it work. 

Only fractions that don't inherit from core would need to add the jandex plugin. But that's just the way it is. I don't think we do much about that.

**Comment by heiko-braun** [comment]
With regard to the choices of 

a) jandex in the fraction plugin and
b) jandex in it's own plugin (like it is now)

the first solution (a) sounds nice, but it would require adding the fraction plugin to any runtime module. currently it's only attached to the main module, but indexes generated at that level don't help. 

the second option (b) seems to be most versatile, doesn't require additional changes to any downstream module and creates an index with every jar. 

**Long Story Short** 

It's complicated, don't ask any further, this is the best approach I could find to balance ease of use with imposed constraints.

**Comment by heiko-braun** [comment]
From my point of view this is ready to be merged

**Comment by bobmcwhirter** [comment]
I'd say merge, and then I'll rebase my :spi stuff and get that in today, hopefully.
